crates/sui-rpc-api/proto/sui.types.proto

```diff
@@ -0,0 +1,1040 @@
+  ObjectId object_id = 1;
+  uint64 version = 2;
+  Digest digest = 3;
+}
+
+message MovePackage {
+  ObjectId id = 1;
+  uint64 version = 2;
+  repeated MoveModule modules = 3;
+  repeated TypeOrigin type_origin_table = 4;
+  repeated UpgradeInfo linkage_table = 5;
+}
+
+message MoveModule {
+  Identifier name = 1;
+  bytes contents = 2;
+}
+
+message TypeOrigin {
+  Identifier module_name = 1;
+  Identifier struct_name = 2;
+  ObjectId package_id = 3;
+}
+
+message UpgradeInfo {
+  ObjectId original_id = 1;
+  ObjectId upgraded_id = 2;
+  uint64 upgraded_version = 3;
+}
+
+message Owner {
+  oneof kind {
+    Address address = 1;
+    ObjectId object = 2;
+    uint64 shared = 3;
+    google.protobuf.Empty immutable = 4;
+  }
+}
+
+message MoveStruct {
+  ObjectId object_id = 1;
+  StructTag object_type = 2;
+  bool has_public_transfer = 3;
+  uint64 version = 4;
+  bytes contents = 5;
+  // rendered
+}
+
+message Object {
+  // Reserve field 1 and 2 for possible bcs bytes and/or Digest of type
+  reserved 1, 2;
+
+  ObjectId object_id = 3;
+  uint64 version = 4;
+  Owner owner = 5;
+
+  ObjectData object = 6;
+
+  Digest previous_transaction = 7;
+  uint64 storage_rebate = 8;
+}
+
+message ObjectData {
+  oneof kind {
+    MoveStruct struct = 1;
+    MovePackage package = 2;
+  }
+}
+
+message GenesisObject {
+  ObjectId object_id = 1;
+  uint64 version = 2;
+  Owner owner = 3;
+  ObjectData object = 4;
+}
+
+// End Object }}}
+
+// GasCostSummary {{{
+
+message GasCostSummary {
+  uint64 computation_cost = 1;
+  uint64 storage_cost = 2;
+  uint64 storage_rebate = 3;
+  uint64 non_refundable_storage_fee = 4;
+}
+
+// End GasCostSummary }}}
+
+// Move {{{
+
+message Identifier {
+  bytes identifier = 1;
+}
+
+message StructTag {
+  Address address = 1;
+  Identifier module = 2;
+  Identifier name = 3;
+  repeated TypeTag type_parameters = 4;
+}
+
+message TypeTag {
+  oneof tag {
+    google.protobuf.Empty U8 = 1;
+    google.protobuf.Empty U16 = 2;
+    google.protobuf.Empty U32 = 3;
+    google.protobuf.Empty U64 = 4;
+    google.protobuf.Empty U128 = 5;
+    google.protobuf.Empty U256 = 6;
+    google.protobuf.Empty Bool = 7;
+    google.protobuf.Empty Address = 8;
+    google.protobuf.Empty Signer = 9;
+    TypeTag vector = 10;
+    StructTag struct = 11;
+  }
+}
+
+message MoveStructValue {
+  StructTag struct_type = 1;
+  repeated MoveField fields = 2;
+}
+
+message MoveField {
+  Identifier name = 1;
+  MoveValue value = 2;
+}
+
+message MoveVariant {
+  StructTag enum_type = 1;
+  Identifier variant_name = 2;
+  uint32 tag = 3;
+  repeated MoveField fields = 4;
+}
+
+message MoveValue {
+  oneof kind {
+    bool bool = 2;
+    uint32 u8 = 3;
+    uint32 u16 = 4;
+    uint32 u32 = 5;
+    uint64 u64 = 6;
+    U128 u128 = 7;
+    U256 u256 = 8;
+    Address address = 9;
+    MoveVector vector = 10;
+    MoveStructValue struct = 11;
+    Address signer = 12;
+    MoveVariant variant = 13;
+  }
+}
+
+message U128 {
+  bytes bytes = 1;
```

Comment:
for these big int types maybe add a comment on what encoding you expect in the bytes field?

also could be good for clarity to put them all together in one place with all the "primitives" vs sprinkled throughout